feature-libs/product-configurator/rulebased/components/price/configurator-price.component.ts

```diff
@@ -1,45 +1,104 @@
-import { Component, Input } from '@angular/core';
-
-export enum ConfiguratorPriceType {
-  PRICE_ONLY = 'priceOnly',
-  QUANTITY_ONLY = 'quantityOnly',
-  PRICE_AND_QUANTITY = 'priceAndQuantity',
-}
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
 
 @Component({
   selector: 'cx-configurator-price',
   templateUrl: './configurator-price.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
As far as I understand, **ChangeDetectionStrategy.OnPush** should be used, otherwise the component will not be updated if the price formula has been changed

Reply:
why? - if the model, such as the price formula changes, angular will detect the changes and re-render.
Actually **https://angular.io/api/core/ChangeDetectionStrategy** deactivates this default behaviour and so there is the need to manually trigger a change detection cycle.